_zephyr.pyx

```diff
@@ -26,12 +27,13 @@ class ZUid(object):
 
 cdef void _ZUid_c2p(ZUnique_Id_t * uid, object p_uid) except *:
     p_uid.address = inet_ntoa(uid.zuid_addr)
-    p_uid.time = uid.tv.tv_sec + (uid.tv.tv_usec / 100000.0)
+    p_uid.time = socket.ntohl(uid.tv.tv_sec) + (socket.ntohl(uid.tv.tv_usec) / 100000.0)
 
 cdef void _ZUid_p2c(object uid, ZUnique_Id_t * c_uid) except *:
     inet_aton(uid.address, &c_uid.zuid_addr)
     c_uid.tv.tv_sec = int(uid.time)
-    c_uid.tv.tv_usec = int((uid.time - c_uid.tv.tv_sec) * 100000)
+    c_uid.tv.tv_usec = socket.htonl(int((uid.time - c_uid.tv.tv_sec) * 100000))
+    c_uid.tv.tv_sec = socket.htonl(c_uid.tv.tv_sec)
```

Comment:
Maybe better to put this as `c_uid.tv.tv_sec = socket.htonl(uid.time)` 2 lines up?

Reply:
I'm using `c_uid.tv.tv_sec` two ways here. First, it holds the seconds in host order for subtraction in the `tv_usec` assignment. Second, it gets the byte order fixed for C. If I flip the byte order sooner, I'll need to cast to an int twice, or use another placeholder.

Reply:
Oh, I see -- yeah I'd just use a new variable for the first entry to make it all more readable.

Reply:
Sure, I can do that.